src/abstract/edwards.ts

```diff
@@ -537,6 +545,7 @@ export function twistedEdwards(curveDef: CurveType): CurveFn {
   G._setWindowSize(8); // Enable precomputes. Slows down first publicKey computation by 20ms.
 
   const utils = {
+    getPrivateScalar,
```

Comment:
Is it necessary to expose it?

Reply:
That's a good question, I'd expose it for two reasons:

1. In rust ([ed25519-dalek](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/hazmat/struct.ExpandedSecretKey.html)) and go ([standard library](https://pkg.go.dev/crypto/internal/fips140/edwards25519@go1.24.1#Scalar.SetBytesWithClamping)), user can get the exposed Scalar
2. Currently we are already exposing `scalar` through `getExtendedPublicKey`, with some extra calls

Reply:
I see. You can easily update it here when you're ready

https://github.com/paulmillr/noble-ed25519/blob/main/index.ts#L245

Reply:
Not really.

ed25519 does not support synchronous hashing out-of-box because it requires no deps. Curves, on the other hand, use noble-hashes.

Because of that, ed25519 provides two implementations: async (using built-in crypto.subtle) and sync (which requires connecting to hashing library first).

So utils will need TWO functions, not one. And it's already a mess with 2 getExtPubkey, 2 getPubkey, 2 sign, 2 verify, etc.

Reply:
I think we may need a better abstraction for the sync and async crypto just like `@noble/ciphers/webcrypto`

Reply:
yeah